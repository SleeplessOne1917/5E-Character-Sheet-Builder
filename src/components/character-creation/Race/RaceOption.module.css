.chevron {
    height: 3rem;
    width: 3rem;
}

.race-container {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: var(--davy);
    color: var(--heaven);
    padding: 1rem;
    cursor: pointer;
    border-radius: 0.5rem;
}

.race-container:focus {
    outline: 10px solid var(--highlight);
}

.race-title {
    font-family: var(--font-fantasy);
    font-size: 1.5rem;
}

.race-li {
    margin: 1rem 0;
}

.subrace-list {
    list-style: none;
}

.subrace-list li {
    margin: 1rem 0;
}

@media screen and (min-width: 768px) {
    .race-title {
        font-size: 2rem;
    }
}