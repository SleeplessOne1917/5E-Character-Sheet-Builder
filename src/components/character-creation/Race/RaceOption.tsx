import {
	ChevronDownIcon,
	ChevronRightIcon,
	ChevronUpIcon
} from '@heroicons/react/solid';
import {
	KeyboardEvent,
	KeyboardEventHandler,
	useCallback,
	useState
} from 'react';
import { SrdItem, SubraceItem } from '../../../types/srd';

import classes from './RaceOption.module.css';

type RaceOptionProps = {
	race: SrdItem;
	subraces?: SubraceItem[];
	onChoose: (subraceIndex?: string) => void;
};

const RaceOption = ({
	race,
	subraces,
	onChoose
}: RaceOptionProps): JSX.Element => {
	const [showSubraces, setShowSubraces] = useState(false);
	const hasMultipleSubraces = subraces && subraces.length > 1;
	const hasOneSubrace = subraces && subraces.length === 1;
	const raceTitle = hasOneSubrace ? subraces[0].name : race.name;

	const handleRaceClick = useCallback(() => {
		if (hasMultipleSubraces) {
			setShowSubraces(prevState => !prevState);
		} else if (hasOneSubrace) {
			onChoose(subraces[0].index);
		} else {
			onChoose();
		}
	}, [setShowSubraces, hasMultipleSubraces, onChoose, hasOneSubrace, subraces]);

	const handleRaceKeyUp: KeyboardEventHandler<HTMLDivElement> = useCallback(
		event => {
			if (event.code === 'Enter') {
				handleRaceClick();
			}
		},
		[handleRaceClick]
	);

	const handleSubraceClick = useCallback(
		(subraceIndex: string) => {
			onChoose(subraceIndex);
		},
		[onChoose]
	);

	const handleSubraceKeyUp = useCallback(
		(event: KeyboardEvent<HTMLDivElement>, subraceIndex: string) => {
			if (event.code === 'Enter') {
				onChoose(subraceIndex);
			}
		},
		[onChoose]
	);

	return (
		<li className={classes['race-li']}>
			<div>
				<div
					className={classes['race-container']}
					tabIndex={0}
					onClick={handleRaceClick}
					onKeyUp={handleRaceKeyUp}
					aria-label={raceTitle}
				>
					<div className={classes['race-title']}>{raceTitle}</div>
					{hasMultipleSubraces ? (
						showSubraces ? (
							<ChevronUpIcon className={classes.chevron} />
						) : (
							<ChevronDownIcon className={classes.chevron} />
						)
					) : (
						<ChevronRightIcon className={classes.chevron} />
					)}
				</div>
				{showSubraces && (
					<ul className={classes['subrace-list']}>
						{subraces?.map(subrace => (
							<li key={subrace.index}>
								<div
									className={classes['race-container']}
									tabIndex={0}
									aria-label={subrace.name}
									onClick={() => handleSubraceClick(subrace.index)}
									onKeyUp={event => handleSubraceKeyUp(event, subrace.index)}
								>
									<div className={classes['race-title']}>{subrace.name}</div>
									<ChevronRightIcon className={classes.chevron} />
								</div>
							</li>
						))}
					</ul>
				)}
			</div>
		</li>
	);
};

export default RaceOption;
