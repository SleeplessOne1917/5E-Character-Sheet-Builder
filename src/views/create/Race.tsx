import classes from "./Race.module.css";

const Race = (): JSX.Element => {
  return (
    <>
      <main className={classes.main}>
        <div className={classes.content}></div>
      </main>
    </>
  );
};

export default Race;
