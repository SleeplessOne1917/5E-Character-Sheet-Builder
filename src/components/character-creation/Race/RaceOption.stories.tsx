import { ComponentMeta, ComponentStory } from '@storybook/react';
import RaceOption, { RaceOptionProps } from './RaceOption';

export default {
	title: 'Components/RaceOption',
	component: RaceOption,
	argTypes: {
		onChoose: { type: 'function' }
	},
	args: {
		iconId: 'dwarf'
	},
	decorators: [story => <ul style={{ listStyle: 'none' }}>{story()}</ul>]
} as ComponentMeta<typeof RaceOption>;

const Template: ComponentStory<typeof RaceOption> = (args: RaceOptionProps) => (
	<RaceOption {...args} />
);

export const OnlyRace = Template.bind({});
OnlyRace.args = {
	race: {
		index: 'dwarf',
		name: 'Dwarf'
	}
};

export const RaceWithOneSubrace = Template.bind({});
RaceWithOneSubrace.args = {
	race: {
		index: 'dwarf',
		name: 'Dwarf'
	},
	subraces: [
		{
			index: 'hill-dwarf',
			name: 'Hill Dwarf',
			race: {
				index: 'dwarf'
			}
		}
	]
};

export const RaceWithMultipleSubraces = Template.bind({});
RaceWithMultipleSubraces.args = {
	race: {
		index: 'dwarf',
		name: 'Dwarf'
	},
	subraces: [
		{
			index: 'hill-dwarf',
			name: 'Hill Dwarf',
			race: {
				index: 'dwarf'
			}
		},
		{
			index: 'mountain-dwarf',
			name: 'Mountain Dwarf',
			race: {
				index: 'dwarf'
			}
		}
	]
};
