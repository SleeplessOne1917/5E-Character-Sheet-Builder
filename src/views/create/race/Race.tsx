import {
	AbilityBonus,
	SrdItem,
	SrdRace,
	SrdSubrace,
	SrdSubraceItem
} from '../../../types/srd';
import { getRace, getSubrace } from '../../../services/raceService';
import { useCallback, useEffect, useState } from 'react';

import ChooseModal from '../../../components/character-creation/ChooseModal/ChooseModal';
import { Descriptor } from '../../../types/creation';
import DescriptorComponent from '../../../components/character-creation/Descriptor/Descriptor';
import LoadingSpinner from '../../../components/LoadingSpinner/LoadingSpinner';
import MainContent from '../../../components/MainContent/MainContent';
import RaceOption from '../../../components/character-creation/Race/RaceOption';
import classes from './Race.module.css';

type RaceProps = {
	races: SrdItem[];
	subraces: SrdSubraceItem[];
};

export const mockRaces = [
	{ index: 'dragonborn', name: 'DragonBorn' },
	{ index: 'dwarf', name: 'Dwarf' },
	{ index: 'elf', name: 'Elf' },
	{ index: 'gnome', name: 'Gnome' },
	{ index: 'half-elf', name: 'Half Elf' },
	{ index: 'half-orc', name: 'Half Orc' },
	{ index: 'halfling', name: 'Halfling' },
	{ index: 'human', name: 'Human' },
	{ index: 'tiefling', name: 'Tiefling' }
];

export const mockSubraces = [
	{ index: 'rock-gnome', name: 'Rock Gnome', race: { index: 'gnome' } },
	{ index: 'hill-dwarf', name: 'Hill Dwarf', race: { index: 'dwarf' } },
	{ index: 'high-elf', name: 'High Elf', race: { index: 'elf' } },
	{
		index: 'lightfoot-halfling',
		name: 'Lightfoot Halfling',
		race: { index: 'halfling' }
	}
];

const reduceAbilityBonuses = (
	bonuses: AbilityBonus[],
	startDesc: string = ''
) =>
	bonuses.reduce(
		(acc, cur) =>
			`${acc}${acc === '' ? '' : ', '}+${cur.bonus} ${
				cur.ability_score.full_name
			}`,
		startDesc
	);

const getAbilityScoreDescription = (race: SrdRace, subrace?: SrdSubrace) => {
	let description: string;

	const allSameBonuses = race.ability_bonuses.reduce(
		(acc: { isSame: boolean; value?: number }, cur) => {
			if (!acc.value) {
				return {
					...acc,
					value: cur.bonus
				};
			} else {
				return {
					...acc,
					isSame: acc.value === cur.bonus
				};
			}
		},
		{ isSame: true, value: undefined }
	);

	if (allSameBonuses.isSame && race.ability_bonuses.length > 1) {
		description = `+${allSameBonuses.value} to ${race.ability_bonuses.reduce(
			(acc, cur, index) =>
				`${acc}${index === race.ability_bonuses.length - 1 ? 'and ' : ''}${
					cur.ability_score.full_name
				}${index === race.ability_bonuses.length - 1 ? '' : ', '}`,
			''
		)}`;
	} else {
		description = reduceAbilityBonuses(race.ability_bonuses);
	}

	if (subrace) {
		description = reduceAbilityBonuses(subrace.ability_bonuses, description);
	}

	if (race.ability_bonus_options) {
		const allSameBonusChoices = race.ability_bonus_options.from.options.reduce(
			(acc: { isSame: boolean; value?: number }, cur) => {
				if (!acc.value) {
					return {
						...acc,
						value: cur.bonus
					};
				} else {
					return {
						...acc,
						isSame: acc.value === cur.bonus
					};
				}
			},
			{ isSame: true, value: undefined }
		);

		if (allSameBonusChoices.isSame) {
			description = `${description}, and +${allSameBonusChoices.value} to ${
				race.ability_bonus_options.choose
			} from ${race.ability_bonus_options.from.options.reduce(
				(acc, cur, index) =>
					`${acc}${
						index === (race.ability_bonus_options?.from.options.length ?? 1) - 1
							? 'and '
							: ''
					}${cur.ability_score.full_name}${
						index === (race.ability_bonus_options?.from.options.length ?? 1) - 1
							? '.'
							: ', '
					}`,
				''
			)}`;
		} else {
			description = `${description}, and ${
				race.ability_bonus_options.choose
			} from ${race.ability_bonus_options.from.options.reduce(
				(acc, cur, index) =>
					`${acc}${
						index === (race.ability_bonus_options?.from.options.length ?? 1) - 1
							? 'and '
							: ''
					}+${cur.bonus} to ${cur.ability_score.full_name}}${
						cur.ability_score.full_name
					}${
						index === (race.ability_bonus_options?.from.options.length ?? 1) - 1
							? '.'
							: ', '
					}`,
				''
			)}`;
		}
	}

	if (!description.includes(' and ') && description.includes(',')) {
		const lastIndexOfComma = description.lastIndexOf(',');
		description =
			description.slice(0, lastIndexOfComma) +
			' and' +
			description.slice(lastIndexOfComma);
	}

	if (description.indexOf(',') === description.lastIndexOf(',')) {
		description = description.replace(/,/g, '');
	}

	if (!description.endsWith('.')) {
		description = description + '.';
	}

	return description;
};

const Race = ({ races, subraces }: RaceProps): JSX.Element => {
	const [error, setError] = useState(false);
	const [loading, setLoading] = useState(false);
	const [showModal, setShowModal] = useState(false);
	const [selectedRace, setSelectedRace] = useState<SrdRace>();
	const [selectedSubrace, setSelectedSubrace] = useState<SrdSubrace>();
	const [consideredRace, setConsideredRace] = useState<SrdRace>();
	const [consideredSubrace, setConsideredSubrace] = useState<SrdSubrace>();
	const [consideredRaceIndex, setConsideredRaceIndex] = useState<string>();
	const [consideredSubraceIndex, setConsideredSubraceIndex] =
		useState<string>();
	const [descriptors, setDescriptors] = useState<Descriptor[]>();

	useEffect(() => {
		if (consideredRace) {
			let theDescriptors: Descriptor[] = [
				{
					title: 'Ability Score Bonuses',
					description: getAbilityScoreDescription(
						consideredRace,
						consideredSubrace
					),
					isOpen: false
				},
				{
					title: 'Age',
					description: consideredRace.age,
					isOpen: false
				},
				{
					title: 'Language',
					description: consideredRace.language_desc,
					isOpen: false
				},
				{
					title: 'Size',
					description: consideredRace.size_description,
					isOpen: false
				},
				{
					title: 'Alignment',
					description: consideredRace.alignment,
					isOpen: false
				}
			];
			theDescriptors = theDescriptors.concat(
				consideredRace.traits.map(trait => ({
					title: trait.name,
					description: trait.desc,
					isOpen: false
				}))
			);

			setDescriptors(theDescriptors);
		} else {
			setDescriptors(undefined);
		}
	}, [consideredRace, setDescriptors, consideredSubrace]);

	useEffect(() => {
		if (consideredRaceIndex) {
			setLoading(true);
			setError(false);
			getRace(consideredRaceIndex).then(result => {
				if (result.data) {
					setConsideredRace(result.data.race);

					if (consideredSubraceIndex) {
						getSubrace(consideredSubraceIndex).then(result => {
							if (result.data) {
								setConsideredSubrace(result.data.subrace);
							} else {
								setError(true);
							}
							setLoading(false);
						});
					} else {
						setLoading(false);
					}
				} else {
					setLoading(false);
					setError(true);
				}
			});
		}
	}, [
		setConsideredRace,
		setConsideredSubrace,
		consideredRaceIndex,
		consideredSubraceIndex,
		setLoading,
		setError
	]);

	const getConsiderRaceHandler = useCallback(
		(raceIndex: string) => {
			return (subraceIndex?: string) => {
				setConsideredRaceIndex(raceIndex);

				if (subraceIndex) {
					setConsideredSubraceIndex(subraceIndex);
				}

				setShowModal(true);
			};
		},
		[setConsideredRaceIndex, setConsideredSubraceIndex, setShowModal]
	);

	const deselectConsideredRace = useCallback(() => {
		setConsideredRace(undefined);
		setConsideredRaceIndex(undefined);
		setConsideredSubrace(undefined);
		setConsideredSubraceIndex(undefined);
	}, [
		setConsideredRace,
		setConsideredRaceIndex,
		setConsideredSubrace,
		setConsideredSubraceIndex
	]);

	const closeModal = useCallback(() => {
		setShowModal(false);
		deselectConsideredRace();
	}, [setShowModal, deselectConsideredRace]);

	const chooseRace = useCallback(() => {
		setSelectedRace(consideredRace);
		setSelectedSubrace(consideredSubrace);
		setShowModal(false);
		deselectConsideredRace();
	}, [
		setSelectedRace,
		consideredRace,
		setSelectedSubrace,
		consideredSubrace,
		setShowModal,
		deselectConsideredRace
	]);

	const toggleDescriptor = useCallback(
		(index: number) => {
			if (descriptors) {
				setDescriptors(prevState =>
					prevState
						? prevState.map((descriptor, i) => {
								if (i === index) {
									return {
										...descriptor,
										isOpen: !descriptor.isOpen
									};
								} else {
									return descriptor;
								}
						  })
						: undefined
				);
			}
		},
		[descriptors, setDescriptors]
	);

	const modalContent = (
		<>
			<h2 className={classes['modal-title']}>
				{loading ? 'Loading...' : error ? 'Error' : consideredRace?.name}
			</h2>
			{loading ? (
				<LoadingSpinner />
			) : error ? (
				<p className={classes['error-message']}>Could not load race details</p>
			) : (
				descriptors &&
				descriptors.map((descriptor, index) => (
					<DescriptorComponent
						key={descriptor.title}
						title={descriptor.title}
						description={descriptor.description}
						isOpen={descriptor.isOpen}
						toggleOpen={() => toggleDescriptor(index)}
					/>
				))
			)}
		</>
	);

	return (
		<>
			<MainContent testId="race">
				<h1 className={classes.title}>Choose Race</h1>
				<ul className={classes['race-list']}>
					{races.map(race => (
						<RaceOption
							race={race}
							subraces={subraces.filter(
								subrace => subrace.race.index === race.index
							)}
							key={race.index}
							onChoose={getConsiderRaceHandler(race.index)}
							iconId={race.index}
							selectable={!showModal}
						/>
					))}
				</ul>
			</MainContent>
			<ChooseModal
				show={showModal}
				mainContent={modalContent}
				iconId={consideredRaceIndex ? consideredRaceIndex : 'custom-race'}
				onChoose={chooseRace}
				onClose={closeModal}
			/>
		</>
	);
};

export default Race;
