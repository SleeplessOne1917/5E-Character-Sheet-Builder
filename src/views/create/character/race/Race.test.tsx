import * as stories from './Race.stories';

import { render, screen } from '@testing-library/react';

import { composeStories } from '@storybook/react';

const { Default } = composeStories(stories);

window.matchMedia = jest.fn().mockImplementation(() => ({
	matches: true,
	addEventListener: jest.fn(),
	removeEventListener: jest.fn()
}));

it('renders correctly', () => {
	render(<Default />);

	expect(screen.getByTestId('race')).toMatchSnapshot();
});
