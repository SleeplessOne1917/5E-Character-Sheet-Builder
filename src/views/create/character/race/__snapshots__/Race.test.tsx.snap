// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  class="main"
  data-testid="race"
>
  <div
    class="content"
  >
    <section
      class="container"
      data-testid="general-info-bar"
    >
      <div
        class="name-container"
      >
        <div
          class="input-and-error-container"
          data-testid="text-input"
        >
          <div
            class="input-and-label-container"
          >
            <div
              class="input-container"
            >
              <input
                class="input"
                id="characterName"
                name="characterName"
                type="text"
                value=""
              />
            </div>
            <label
              class="label light-label"
              for="characterName"
            >
              Character Name
            </label>
          </div>
        </div>
      </div>
      <div
        aria-label="Set Hit Points"
        class="hp-container"
        data-testid="hit-points-button"
        role="button"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="hp-icon"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.645 20.91l-.007-.003-.022-.012a15.247 15.247 0 01-.383-.218 25.18 25.18 0 01-4.244-3.17C4.688 15.36 2.25 12.174 2.25 8.25 2.25 5.322 4.714 3 7.688 3A5.5 5.5 0 0112 5.052 5.5 5.5 0 0116.313 3c2.973 0 5.437 2.322 5.437 5.25 0 3.925-2.438 7.111-4.739 9.256a25.175 25.175 0 01-4.244 3.17 15.247 15.247 0 01-.383.219l-.022.012-.007.004-.003.001a.752.752 0 01-.704 0l-.003-.001z"
          />
        </svg>
        <div
          class="hp"
        >
          â€” HP
        </div>
      </div>
    </section>
    <h1
      class="title"
    >
      Choose Race
    </h1>
    <ul
      class="race-list"
    >
      <li
        class="option-li"
        data-testid="option"
      >
        <div>
          <div
            aria-label="DragonBorn"
            class="option-container"
            role="button"
            tabindex="0"
          >
            <div
              class="option-title"
            >
              <div
                class="icon-container"
              >
                <svg
                  class="icon"
                >
                  <use
                    xlink:href="/Icons.svg#dragonborn"
                  />
                </svg>
              </div>
               
              DragonBorn
            </div>
            <svg
              aria-hidden="true"
              class="chevron"
              data-testid="chevron-right"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M16.28 11.47a.75.75 0 010 1.06l-7.5 7.5a.75.75 0 01-1.06-1.06L14.69 12 7.72 5.03a.75.75 0 011.06-1.06l7.5 7.5z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </li>
      <li
        class="option-li"
        data-testid="option"
      >
        <div>
          <div
            aria-label="Hill Dwarf"
            class="option-container"
            role="button"
            tabindex="0"
          >
            <div
              class="option-title"
            >
              <div
                class="icon-container"
              >
                <svg
                  class="icon"
                >
                  <use
                    xlink:href="/Icons.svg#dwarf"
                  />
                </svg>
              </div>
               
              Hill Dwarf
            </div>
            <svg
              aria-hidden="true"
              class="chevron"
              data-testid="chevron-right"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M16.28 11.47a.75.75 0 010 1.06l-7.5 7.5a.75.75 0 01-1.06-1.06L14.69 12 7.72 5.03a.75.75 0 011.06-1.06l7.5 7.5z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </li>
      <li
        class="option-li"
        data-testid="option"
      >
        <div>
          <div
            aria-label="High Elf"
            class="option-container"
            role="button"
            tabindex="0"
          >
            <div
              class="option-title"
            >
              <div
                class="icon-container"
              >
                <svg
                  class="icon"
                >
                  <use
                    xlink:href="/Icons.svg#elf"
                  />
                </svg>
              </div>
               
              High Elf
            </div>
            <svg
              aria-hidden="true"
              class="chevron"
              data-testid="chevron-right"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M16.28 11.47a.75.75 0 010 1.06l-7.5 7.5a.75.75 0 01-1.06-1.06L14.69 12 7.72 5.03a.75.75 0 011.06-1.06l7.5 7.5z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </li>
      <li
        class="option-li"
        data-testid="option"
      >
        <div>
          <div
            aria-label="Rock Gnome"
            class="option-container"
            role="button"
            tabindex="0"
          >
            <div
              class="option-title"
            >
              <div
                class="icon-container"
              >
                <svg
                  class="icon"
                >
                  <use
                    xlink:href="/Icons.svg#gnome"
                  />
                </svg>
              </div>
               
              Rock Gnome
            </div>
            <svg
              aria-hidden="true"
              class="chevron"
              data-testid="chevron-right"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M16.28 11.47a.75.75 0 010 1.06l-7.5 7.5a.75.75 0 01-1.06-1.06L14.69 12 7.72 5.03a.75.75 0 011.06-1.06l7.5 7.5z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </li>
      <li
        class="option-li"
        data-testid="option"
      >
        <div>
          <div
            aria-label="Half Elf"
            class="option-container"
            role="button"
            tabindex="0"
          >
            <div
              class="option-title"
            >
              <div
                class="icon-container"
              >
                <svg
                  class="icon"
                >
                  <use
                    xlink:href="/Icons.svg#half-elf"
                  />
                </svg>
              </div>
               
              Half Elf
            </div>
            <svg
              aria-hidden="true"
              class="chevron"
              data-testid="chevron-right"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M16.28 11.47a.75.75 0 010 1.06l-7.5 7.5a.75.75 0 01-1.06-1.06L14.69 12 7.72 5.03a.75.75 0 011.06-1.06l7.5 7.5z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </li>
      <li
        class="option-li"
        data-testid="option"
      >
        <div>
          <div
            aria-label="Half Orc"
            class="option-container"
            role="button"
            tabindex="0"
          >
            <div
              class="option-title"
            >
              <div
                class="icon-container"
              >
                <svg
                  class="icon"
                >
                  <use
                    xlink:href="/Icons.svg#half-orc"
                  />
                </svg>
              </div>
               
              Half Orc
            </div>
            <svg
              aria-hidden="true"
              class="chevron"
              data-testid="chevron-right"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M16.28 11.47a.75.75 0 010 1.06l-7.5 7.5a.75.75 0 01-1.06-1.06L14.69 12 7.72 5.03a.75.75 0 011.06-1.06l7.5 7.5z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </li>
      <li
        class="option-li"
        data-testid="option"
      >
        <div>
          <div
            aria-label="Lightfoot Halfling"
            class="option-container"
            role="button"
            tabindex="0"
          >
            <div
              class="option-title"
            >
              <div
                class="icon-container"
              >
                <svg
                  class="icon"
                >
                  <use
                    xlink:href="/Icons.svg#halfling"
                  />
                </svg>
              </div>
               
              Lightfoot Halfling
            </div>
            <svg
              aria-hidden="true"
              class="chevron"
              data-testid="chevron-right"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M16.28 11.47a.75.75 0 010 1.06l-7.5 7.5a.75.75 0 01-1.06-1.06L14.69 12 7.72 5.03a.75.75 0 011.06-1.06l7.5 7.5z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </li>
      <li
        class="option-li"
        data-testid="option"
      >
        <div>
          <div
            aria-label="Human"
            class="option-container"
            role="button"
            tabindex="0"
          >
            <div
              class="option-title"
            >
              <div
                class="icon-container"
              >
                <svg
                  class="icon"
                >
                  <use
                    xlink:href="/Icons.svg#human"
                  />
                </svg>
              </div>
               
              Human
            </div>
            <svg
              aria-hidden="true"
              class="chevron"
              data-testid="chevron-right"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M16.28 11.47a.75.75 0 010 1.06l-7.5 7.5a.75.75 0 01-1.06-1.06L14.69 12 7.72 5.03a.75.75 0 011.06-1.06l7.5 7.5z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </li>
      <li
        class="option-li"
        data-testid="option"
      >
        <div>
          <div
            aria-label="Tiefling"
            class="option-container"
            role="button"
            tabindex="0"
          >
            <div
              class="option-title"
            >
              <div
                class="icon-container"
              >
                <svg
                  class="icon"
                >
                  <use
                    xlink:href="/Icons.svg#tiefling"
                  />
                </svg>
              </div>
               
              Tiefling
            </div>
            <svg
              aria-hidden="true"
              class="chevron"
              data-testid="chevron-right"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M16.28 11.47a.75.75 0 010 1.06l-7.5 7.5a.75.75 0 01-1.06-1.06L14.69 12 7.72 5.03a.75.75 0 011.06-1.06l7.5 7.5z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </li>
    </ul>
  </div>
</main>
`;
